x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java

```diff
@@ -1593,7 +1593,16 @@ public enum Cap {
          * Fix management of plans with no columns
          * https://github.com/elastic/elasticsearch/issues/120272
          */
-        FIX_NO_COLUMNS;
+        FIX_NO_COLUMNS,
+
+        /**
+         * Fix attribute equality to respect the name id of the attribute.
+         */
+        ATTRIBUTE_EQUALS_RESPECTS_NAME_ID,
```

Comment:
Move this up, we should avoid adding capabilities to the end of the enum.

Reply:
FWIW, I like this solution. Having caps in order makes sense to understand what was created first, and it will make our life easier when we decide to prune some of them.